app/controllers/rsvps_controller.rb

```diff
@@ -27,9 +27,13 @@ def create
     end
 
     Rsvp.transaction do
-      if @event.at_limit? && @rsvp.role_student?
+      if @event.students_at_limit? && @rsvp.role_student?
         @rsvp.waitlist_position = (@event.rsvps.maximum(:waitlist_position) || 0) + 1
```

Comment:
I think this needs to find the maximum waitlist position for **student waitlist RSVPs** only, since the waitlist sizes / positions will differ between students and volunteers.